nucypher-core-wasm/src/lib.rs

```diff
@@ -75,12 +76,13 @@ where
 // TODO (rust-umbral#23): This is necessary since wasm-bindgen does not support
 // having a parameter of `Vec<&T>`
 // (see https://github.com/rustwasm/wasm-bindgen/issues/111).
-fn try_from_js_array<T>(value: &JsValue) -> Result<Vec<T>, Error>
+fn try_from_js_array<T>(value: impl AsRef<JsValue>) -> Result<Vec<T>, Error>
```

Comment:
What does `impl` do here?

Reply:
It is a shorthand for an equivalent declaration
```rust
fn try_from_js_array<T, U: impl AsRef<JsValue>>(value: U) -> Result<Vec<T>, Error>
```
If we only ever use `U` in one argument, and the constraint is short enough, it is conventional to just write the constraint directly as a type.